src/core/src/signature.rs

```diff
@@ -107,19 +110,33 @@ pub trait SigsTrait {
 
                 aa.windows(aa_ksize as usize).for_each(|n| {
                     let hash = crate::_hash_murmur(n, self.seed());
-                    self.add_hash(hash);
+                    seq_hashes.push(hash);
                 });
 
                 let rc_substr: Vec<u8> = rc.iter().cloned().skip(i).take(rc.len() - i).collect();
                 let aa_rc = to_aa(&rc_substr, hash_function.dayhoff(), hash_function.hp()).unwrap();
 
                 aa_rc.windows(aa_ksize as usize).for_each(|n| {
                     let hash = crate::_hash_murmur(n, self.seed());
-                    self.add_hash(hash);
+                    seq_hashes.push(hash);
                 });
             }
         }
 
+        Ok(seq_hashes)
+    }
+
+    fn add_sequence(&mut self, seq: &[u8], force: bool) -> Result<(), Error> {
+        let ready_hashes = match self.seq_to_hashes(seq, force) {
+            Ok(s) => s,
+            Err(e) => return Err(e),
+        };
+
+        for hash_value in ready_hashes.iter() {
```

Comment:
If you're using `for` loops it is usually not needed to call `.iter()`, because using `&ready_hashes` does that:
```suggestion
        for hash_value in &ready_hashes {
```

(for completeness: `for h in ready_hashes` iterates by value, consuming the vector, and is equivalent to call `ready_hashes.into_iter()`)

Reply:
Thanks for explaining that :)